.github/workflows/main.yml

```diff
@@ -0,0 +1,107 @@
+# This GitHub Actions workflow contains four jobs:
+# -- First, Lint Code uses the latest Ubuntu image and Python 3.9 to check
+# the code for defects, using the Poetry lint task.
+# -- Second, Lint Markdown uses the latest Ubuntu image and makrdownlint to
+# check markdown files for defects.
+# -- Third uses the latest Ubuntu image and cspell to check for spelling
+# defects in the markdown.
+# -- Fourth, Test uses a strategy matrix to test the code with different
+# Python versions and determine code coverage, using the Poetry test task.
+
+name: Lint and Test
+
+on: [push, pull_request]
+
+jobs:
+  code:
+    name: Lint Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v2
+        id: setup-python
+        with:
+          python-version: 3.9
+      - name: Install Poetry
+        uses: Gr1N/setup-poetry@v7
+      - name: Use Poetry Dependency Cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pypoetry/virtualenvs
+          key: ${{ runner.os }}-poetry-${{ hashFiles('poetry.lock') }}
+      - name: Setup Poetry
+        run: |
+          poetry --version
+          poetry config virtualenvs.create true
+          poetry config virtualenvs.in-project true
+          poetry env info
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-ansi
+      - name: Lint code
+        run: poetry run task lint
+  markdown:
+    name: Lint Markdown
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Run markdownlint
+        uses: DavidAnson/markdownlint-cli2-action@v4
+  spelling:
+      name: Check Spelling
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+          with:
+            fetch-depth: 0
+        - name: Run cspell
+          uses: zwaldowski/cspell-action@v1
+          with:
+            paths: ["*.md", "*/*.md"] #Not sure if multiple paths are supported?
+            config: .github/cspell.json
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
To keep things sparse (and have jobs run quickly), it may be a good idea to not include Python 3.9 and 3.8 for testing on Linux.